test/unit/clock_lowres.c

```diff
@@ -0,0 +1,34 @@
+#include "unit.h"
+
+#include <math.h>
+#include "clock_lowres.h"
+#include "clock.h"
+
+/** Test duration in seconds. */
+#define TEST_LEN 1.5
+
+int
+main(void)
+{
+	plan(1);
+	clock_lowres_signal_init();
+
+	/* Such a large resolution to pass the test in debug or apple build. */
+	double resolution = clock_lowres_resolution() * 8;
```

Comment:
Doesn't seem to help - the test still fails on OSX 15:

[![image](https://user-images.githubusercontent.com/1114511/171197652-6dbd43c5-0ffb-4be9-8b72-8188a8629ada.png)](https://github.com/tarantool/tarantool/runs/6671894055?check_suite_focus=true)

Reply:
Seems like x10 helps

Reply:
It looks like the failure happened because the test checked the clock in a busy loop so the process consumed all its quota and was rescheduled in the middle of a check. Let's sleep for `resolution` after each check. This way, the process won't consume a lot of CPU time and therefore won't be laid off CPU for too long. Then we can assume that the max deviation is no more than `resolution * 2`.